units/src/parse.rs

```diff
@@ -114,6 +114,18 @@ pub fn hex_u32<S: AsRef<str> + Into<String>>(s: S) -> Result<u32, ParseIntError>
     })
 }
 
+/// Parses a u128 from a hex string.
+///
+/// Input string may or may not contain a `0x` prefix.
```

Comment:
Is this true? Looking at code, this only works for unprefixed outputs.

Reply:
You are correct, not sure how it got to this state - `hex_u32` and `hex_u128` are different, they should behave the same, I'll have to investigate more to work out what we want. Good reviewing, thanks.